### PR TITLE
[JENKINS-44830] NPE after upgrading caused by fix of JENKINS-42959

### PR DESCRIPTION
**Pull request by jglick**
[JENKINS-44830](https://issues.jenkins-ci.org/browse/JENKINS-44830)

Amends #54. Reproduces reported NPE.

@reviewbybees

### CODE REVIEW

File: src/main/java/hudson/plugins/sshslaves/SSHLauncher.java [hunk 1]
```diff
@@ -553,10 +554,16 @@ public String getCredentialsId() {
         return credentialsId;
     }
 
+    @CheckForNull
     public SshHostKeyVerificationStrategy getSshHostKeyVerificationStrategy() {
         return sshHostKeyVerificationStrategy;
     }
 
+    @NonNull
+    SshHostKeyVerificationStrategy getSshHostKeyVerificationStrategyDefaulted() {
```
Comment (oleg-nenashev) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: makes sense to make public later

Comment (jglick) [inline src/main/java/hudson/plugins/sshslaves/SSHLauncher.java]: Only if there are some external callers—currently I do not see a need for it.

**Comment by oleg-nenashev** [comment]
@reviewbybees done

**Comment by jglick** [comment]
> Does this add a requirement to build on Linux that wasn't there before?

Nope, if the `docker` command is not present, the test just gets skipped (as it does on the currently configured CI job).

**Comment by rtyler** [comment]
[thanks jesse](https://www.youtube.com/watch?v=ZkvMRROkbC0&t=83)

**Pull request merged**